osquery/tables/system/windows/registry.cpp

```diff
@@ -151,14 +166,16 @@ void queryKey(const std::string& keyPath, QueryData& results) {
     return;
   }
 
-  BYTE* bpDataBuff = new BYTE[cbMaxValueData];
   DWORD cchValue = maxKeyLength;
   TCHAR achValue[maxValueName];
 
   // Process registry values
   for (size_t i = 0, retCode = ERROR_SUCCESS; i < cValues; i++) {
+    DWORD lpData = (cbMaxValueData == 0) ? NULL : cbMaxValueData;
```

Comment:
Any reason these three variables were brought "up"? Maybe push them down closer to where they're used in the off change the `RegEnumValue` fails they're not allocated and left unused?